Source/WebCore/accessibility/AXTextMarker.cpp

```diff
@@ -858,6 +941,53 @@ AXTextMarkerRange AXTextMarker::lineRange(LineRangeType type) const
     RELEASE_ASSERT_NOT_REACHED();
 }
 
+AXTextMarkerRange AXTextMarker::wordRange(WordRangeType type) const
+{
+    if (!isValid())
+        return { { }, { } };
+
+    AXTextMarker startMarker, endMarker;
+
+    if (type == WordRangeType::Right) {
+        startMarker = nextWordStart();
+        endMarker = startMarker.nextWordEnd();
+        auto resultRange = AXTextMarkerRange { startMarker, endMarker };
+        auto resultRangeString = resultRange.toString();
+
+        // A newline should not be returned as the right word (if we are at the end of a paragraph)
+        if (startMarker == *this && resultRangeString.length() && resultRangeString[0] == '\n')
+            return { WTFMove(*this), WTFMove(*this) };
+    } else {
+        startMarker = previousWordStart();
+        if (startMarker == *this)
+            return { startMarker, startMarker };
+
+        endMarker = startMarker.nextWordEnd();
+        auto resultRange = AXTextMarkerRange { startMarker, endMarker };
+        auto resultRangeString = resultRange.toString();
```

Comment:
Here's another spot where it could be nice to avoid unnecessarily calling `AXTextMarkerRange::toString()` — seems like we only need it when `equivalentTextPosition(endMarker)` is true.